### ISSUE TITLE
Plugin feature with function variables

### ISSUE THREAD

**Issue by felix-r** [ZELLMECHANIK-DRESDEN/dclab#127]
I would like to compute a plugin feature from a function that needs more variables than the ones available in a dclab dataset. From the docs I could not really find out how to handle this case but I have the feeling there is some pretty pythonian way to solve this that I'm just not able to come up with. To illustrate a little more, I would like to do something like:

```python
def new_features(rtdc_ds, var_1, var_2):
    circ_per_area = var_1 * rtdc_ds["circ"] / rtdc_ds["area_um"]
    circ_times_area = var_2 * rtdc_ds["circ"] * rtdc_ds["area_um"]
    # returns a dictionary-like object
    return {"circ_per_area": circ_per_area, "circ_times_area": circ_times_area}

info = {
    "method": new_features,
    "description": "This plugin will compute some features",
    "long description": "Even longer description that "
                        "can span multiple lines",
    "feature names": ["circ_per_area", "circ_times_area"],
    "feature labels": ["Circularity per Area", "Circularity times Area"],
    "features required": ["circ", "area_um"],
    "config required": [],
    "method check required": lambda x: True,
    "scalar feature": [True, True],
    "version": "0.1.0",
}
```

How can I load this with `dclab.load_plugin_feature("/path/to/plugin.py")`  and also pass `var_1, var_2` to the function call?

**Comment by paulmueller** [comment]
It really depends on what you are trying to do. Where do the variables come from?

PluginFeatures are designed to be globally applicable. They require you to pass all necessary information via `rtdc_ds` (RTDCBase object). Can you infer the variables from anything (e.g. the path) of the dataset?

What about user-defined metadata (#125 ) - Would that help?

**Comment by felix-r** [comment]
Yes, user-defined metadata would help in my case.

**Comment by paulmueller** [comment]
As a general answer to this question:

If you can write a function that gets the additional metadata for a specific dataset from an external source (e.g. from an excel sheet containing the path of the dataset as a column) or if you can rely on user-defined features, then you are good.

Otherwise, PluginFeatures are not for you.

**Comment by PinkShnack** [comment]
For clarification, one can now add user-defined metadata to dclab datasets. This was implemented in #129. 

See [the docs for information](https://dclab.readthedocs.io/en/latest/sec_av_notation.html#user-defined-metadata).
